crates/chain-listener/src/listener.rs

```diff
@@ -878,44 +864,54 @@ impl ChainListener {
             None => return Ok(()),
         };
 
-        let priority_units = self.get_priority_units();
-        let non_priority_units = self.get_non_priority_units();
-        let pending_units = self.get_pending_units();
+        let group_result = self.get_cu_groups();
+
+        let cc_cores = self.acquire_cores_for_cc(&group_result)?;
 
-        let mut cu_allocation = HashMap::new();
-        let priority_cores = self.acquire_cores_for_cc(&priority_units)?;
-        let non_priority_cores = self.acquire_cores_for_cc(&non_priority_units)?;
-        let pending_cores = self.acquire_cores_for_cc(&pending_units)?;
+        let mut cu_allocation: HashMap<PhysicalCoreId, CUID> = HashMap::new();
 
-        if all_min_proofs_found(&priority_units) {
+        if group_result.all_min_proofs_found() {
             tracing::info!(target: "chain-listener", "All CUs found minimal number of proofs {} in current epoch {}", self.min_proofs_per_epoch, self.current_epoch);
-            if all_max_proofs_found(&non_priority_units) {
+            if group_result.all_max_proofs_found() {
                 tracing::info!(target: "chain-listener", "All CUs found max number of proofs {} in current epoch {}", self.max_proofs_per_epoch ,self.current_epoch);
                 self.stop_commitment().await?;
                 return Ok(());
             } else {
                 // All CUs were proven, now let's work on submitting proofs for every CU until MAX_PROOF_COUNT is reached
-                cu_allocation.extend(non_priority_cores.iter().zip(non_priority_units.iter()));
-
-                let mut units = non_priority_units.iter().cycle();
+                cu_allocation.extend(
+                    cc_cores
+                        .non_priority_cores
+                        .iter()
+                        .cloned()
+                        .zip(group_result.non_priority_units.iter().cloned()),
+                );
+
+                let mut units = group_result.non_priority_units.iter().cycle();
                 // Assign "pending cores" to help generate proofs for "non priority units"
```

Comment:
It took me 10-20 minutes to understand what's happening here :( Names are super hard to understand.

```suggestion
                // Assign "pending cores" to help generate proofs for "non priority units"
                let mut units = group_result.non_priority_units.iter().cycle();
```

Reply:
done